app-frontend/src/app/components/datasources/datasourceItem/datasourceItem.html

```diff
@@ -30,10 +30,17 @@
         type="button"
         class="btn btn-small btn-danger"
         ng-click="$ctrl.onOpenDatasourceDeleteModal()"
-        ng-if="$ctrl.isOwner && !$ctrl.isShared"
-        uib-tooltip="After today restriction">
+        ng-if="$ctrl.isOwner"
+        ng-class="{'btn-danger-disabled': $ctrl.isShared}">
         <span class="sr-only">Delete</span>
-        <i class="icon-trash"></i>
+        <i class="icon-trash" ng-if="!$ctrl.isShared"></i>
+        <i class="icon-trash"
+           ng-if="$ctrl.isShared"
+           tooltips
+           tooltip-template="Datasource cannot be deleted at this time because it is being shared."
```

Comment:
tooltips break pretty much every native angular directive.
To get around it, stick the tooltip on a container element for the element that needs directives.